contracts/0.8.25/vaults/dashboard/NodeOperatorFee.sol

```diff
@@ -6,35 +6,54 @@ pragma solidity 0.8.25;
 
 import {Math256} from "contracts/common/lib/Math256.sol";
 
-import {IStakingVault} from "./interfaces/IStakingVault.sol";
-import {Dashboard} from "./Dashboard.sol";
+import {IStakingVault} from "../interfaces/IStakingVault.sol";
+import {Permissions} from "./Permissions.sol";
+import {VaultHub} from "../VaultHub.sol";
 
 /**
- * @title Delegation
- * @notice This contract is a contract-owner of StakingVault and includes an additional delegation layer.
+ * @title NodeOperatorFee
+ * @notice This contract manages the node operator fee and claiming mechanism.
+ * It reserves a portion of the staking rewards for the node operator, and allows
+ * the node operator to claim their fee.
+ *
+ * Key features:
+ * - Tracks node operator fees based on staking rewards
+ * - Provides fee claiming mechanism via role-based access control
+ * - Restricts withdrawals based on the fee reserved for node operators
+ * - Requires both the node operator and default admin to confirm the fee changes
+ *
+ * Node operator fees are calculated as a percentage of the staking rewards accrued
+ * between the last claimed report and the latest report in the StakingVault.
+ * If the fee was never claimed, the percentage is calculated based on the total
+ * rewards accrued since the StakingVault was created.
  */
-contract Delegation is Dashboard {
+abstract contract NodeOperatorFee is Permissions {
     /**
-     * @notice Maximum combined feeBP value; equals to 100%.
+     * @notice Total basis points; 1bp = 0.01%, 100_00bp = 100.00%.
+     */
+    uint256 private constant TOTAL_BASIS_POINTS = 100_00;
+
+    /**
+     * @notice Maximum fee value; equals to 100.00%.
      */
     uint256 private constant MAX_FEE_BP = TOTAL_BASIS_POINTS;
```

Comment:
🤔 why do we need separate private const? maybe use TOTAL_BASIS_POINTS instead?

Reply:
For me, the issue lies in usage. MAX_FEE_BP suggests it could be less than or greater than 100%, yet the error states it is always 100%

<img width="571" alt="Screenshot 2025-04-09 at 16 20 45" src="https://github.com/user-attachments/assets/603845c4-af13-487d-88e5-9b16c9adc696" />